components/utils.py

```diff
@@ -1004,20 +995,137 @@ def _get_rollback_resources_dir(self, service_name):
                             service_name,
                             self.RESOURCES_ROLLBACK_DIR_NAME)
 
+    def _get_rollback_resources_json(self, service_name):
+        rollback_dir = self._get_rollback_resources_dir(service_name)
+        rollback_json = os.path.join(rollback_dir, self.RESOURCES_JSON_FILE)
+        with open(rollback_json) as f:
+            return json.load(f)
+
 
 resource_factory = BlueprintResourceFactory()
 ctx_factory = CtxPropertyFactory()
 
 
-def start_service_and_archive_properties(service_name,
-                                         append_prefix=True,
-                                         ignore_restart_fail=False):
-
+def start_service(service_name, append_prefix=True, ignore_restart_fail=False):
     if is_upgrade:
         systemd.restart(service_name,
                         ignore_failure=ignore_restart_fail,
                         append_prefix=append_prefix)
-        ctx_factory.archive_properties(service_name)
-        resource_factory.archive_resources(service_name)
     else:
         systemd.start(service_name, append_prefix=append_prefix)
+
+
+def http_request(url, data=None, method='PUT', headers={}):
+    request = urllib2.Request(url, data=data, headers=headers)
+    request.get_method = lambda: method
+    try:
+        return urllib2.urlopen(request)
+    except urllib2.URLError as e:
+        reqstring = url + (' ' + data if data else '')
+        ctx.logger.error('Failed to {0} {1} (reason: {2})'.format(
+                method, reqstring, e.reason))
+
+
+def _wait_for_execution(execution_id, headers):
+    poll_interval = 2
+    url = 'http://localhost/api/v2/executions?' \
+          '_include_system_workflows=true&id={0}'.format(execution_id)
+    while True:
```

Comment:
So you think we should put a timeout on the snapshot restore process? This process duration really depends on how many deployments a user may have. What timeout would you give it?